rye/src/cli/publish.rs

```diff
@@ -157,10 +141,11 @@ pub fn execute(cmd: Args) -> Result<(), Error> {
         .args(files)
         .arg("--username")
         .arg(username)
-        .arg("--password")
-        .arg(token.expose_secret())
         .arg("--repository-url")
         .arg(repository_url.to_string());
+    if let Some(secret) = maybe_secret {
```

Comment:
the other key bit. we only add a `password` if a secret got created, ie if there was a non-empty string (or some credential looked up from file)